src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksNewVendorRequestsServiceImpl.java

```diff
@@ -0,0 +1,412 @@
+              getPaymentWorksWebServiceCallsService().sendRejectedStatusToPaymentWorksForNewVendor(stgNewVendorRequestDetailToProcess.getPmwVendorRequestId());
+            }
+        }
+        getPaymentWorksNewVendorRequestsReportService().generateAndEmailProcessingReport(reportData);
+    }
+    
+    private boolean pmwNewVendorRequestProcessingIntoKfsWasSuccessful(PaymentWorksVendor savedStgNewVendorRequestDetailToProcess, PaymentWorksNewVendorRequestsBatchReportData reportData) {
+        if (getPaymentWorksDataProcessingIntoKfsService().ableToCreateValidateRouteKfsVendor(savedStgNewVendorRequestDetailToProcess, getPaymentWorksIsoToFipsCountryMap(), getPaymentWorksToKfsDiversityMap(), reportData)) { 
+            updatePmwNewVendorStagingTableBasedOnSuccessfulCreateVendorProcessing(savedStgNewVendorRequestDetailToProcess);
+            updatePmwNewVendorReportBasedOnSuccessfulCreateVendorProcessing(savedStgNewVendorRequestDetailToProcess, reportData);
+            return true;
+        }
+        else {
+            updatePmwNewVendorStagingTableBasedOnCreateVendorFailing(savedStgNewVendorRequestDetailToProcess);
+            return false;
+        }
+    }
+    
+    private boolean canPaymentWorksNewVendorRequestProcessingContinueForVendor(PaymentWorksVendor stgNewVendorRequestDetailToProcess, PaymentWorksNewVendorRequestsBatchReportData reportData) {
+        List<String> errorMessages = new ArrayList<String>();
+        if (pmwDtosCouldConvertCustomAttributesToPmwJavaClassAttributes(stgNewVendorRequestDetailToProcess) &&
+            pmwNewVendorAttributesConformToKfsLengthsOrFormats(stgNewVendorRequestDetailToProcess, errorMessages) &&
+            allPmwNewVendorIsoCountriesMapToSingleFipsCountry(stgNewVendorRequestDetailToProcess, errorMessages) &&
+            pmwNewVendorIdentifierDoesNotExistInKfsStagingTable(stgNewVendorRequestDetailToProcess, errorMessages)){
+            return true;
+        }
+        else {
+            if (!errorMessages.isEmpty()){
+                reportData.getPendingPaymentWorksVendorsThatCouldNotBeProcessed().incrementRecordCount();
+                reportData.addPmwVendorsThatCouldNotBeProcessed(new PaymentWorksBatchReportRawDataItem(stgNewVendorRequestDetailToProcess.toString(), errorMessages));
+            }
+            return false;
+        }
+    }
+    
+    private boolean pmwDtosCouldConvertCustomAttributesToPmwJavaClassAttributes(PaymentWorksVendor stgNewVendorRequestDetailToProcess) {
+        if (stgNewVendorRequestDetailToProcess.isCustomFieldConversionErrors()) {
+            return false;
+        }
+        else{
+            return true;
+        }
+    }
+    
+    private boolean pmwNewVendorAttributesConformToKfsLengthsOrFormats(PaymentWorksVendor stgNewVendorRequestDetailToProcess, List<String> errorMessages) {
+        boolean allValidationPassed = true;
+        if (StringUtils.isNotBlank(stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalLastName()) &
```

Comment:
could this be extracted in a separate method?

Reply:
Changed as requested.